### ISSUE TITLE
Add support for scheduling_strategy on aws_ecs_service

### ISSUE THREAD

**Issue by mwarkentin** [terraform-providers/terraform-provider-aws#4820]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Support for `daemon` services (1 task per ECS instance) was just added to ECS as a `schedulingStrategy` attribute when creating a service ([api docs](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_CreateService.html#ECS-CreateService-request-schedulingStrategy)). Valid values are:

* `REPLICA`: this is the strategy that all ECS tasks used to be
* `DAEMON`: one task per ECS instance

It would be great to be able to create services of this type with terraform.

### New or Affected Resource(s)

* aws_ecs_service

### Potential Terraform Configuration

Note: I'm not sure if the `scheduling_strategy` values should be lowercase like most terraform configs or UPPERCASE to match the AWS API values.

```hcl
resource "aws_ecs_service" "datadog" {
  name                = "datadog"
  cluster             = "${aws_ecs_cluster.foo.id}"
  task_definition     = "${aws_ecs_task_definition.datadog.arn}"
  iam_role            = "${aws_iam_role.foo.arn}"
  scheduling_strategy = "daemon"
}
```

### References

* https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_CreateService.html#ECS-CreateService-request-schedulingStrategy

**Comment by SaikiranDaripelli** [comment]
Any update on when this will be merged?

**Comment by bflad** [comment]
We should be able to get this released tomorrow, more shortly.

**Comment by bflad** [comment]
Support for the new `scheduling_strategy` argument and `DAEMON` ECS services has been merged in and will release with version 1.25.0 of the AWS provider tomorrow.

**Comment by bflad** [comment]
This has been released in [version 1.25.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1250-june-27-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by billcrook** [comment]
Thanks for this! Already using it in prod.
